streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -142,6 +146,25 @@ public StreamTask(final TaskId id,
     }
 
     /**
+     * Finds the source of the given topic partition.
+     *
+     * @param partition the topic partition
+     * @return
+     */
+    private SourceNode findSource(final TopicPartition partition) {
```

Comment:
@guozhangwang I fixed the tests as well and added this code snippet too. 
I mentioned this in 
https://github.com/apache/kafka/pull/2466#issuecomment-282328998